src/msg/async/ProtocolV2.cc

```diff
@@ -2275,20 +2287,18 @@ CtPtr ProtocolV2::send_reconnect() {
   ldout(cct, 20) << __func__ << dendl;
 
   ReconnectFrame reconnect(this, messenger->get_myaddrs(),
-			   connection->target_addr,
-			   cookie,
-			   messenger->get_myname().num(),
-			   global_seq,
+                           client_cookie,
+			                     server_cookie,
+			                     global_seq,
                            connect_seq,
-			   connection->policy.features_supported,
-			   connection->policy.features_required | msgr2_required,
-			   in_seq);
+			                     in_seq);
```

Comment:
I don't understand why the id and features are removed.  Looking at 91a888ecb46aae4119eb38214cc7b86c86aab6bc, which solved a specific case of

    - A connects to B
    - A sends client_ident
    - fault before A gets server_ident, so A doesn't know B's features or name
    - B reconnects to A
    - connection established

Reply:
Because now, when B reconnects to A, A will tell B to soft reset (clean its server_cookie) and B sends a client_ident with the same client_cookie. Then A will know the client_cookie and resumes the session with a new server_cookie by sending a server_ident to B.

With the two cookies is now possible to know if we were interrupted between the client_ident <-> server_ident exchange, and if one of the peers tries to reconnect with a "reconnect" Tag, I'm falling back to the client_ident <-> server_ident exchange.

Reply:
Excellent.  My only question then is if we can make this clearer with a different tag name, like TAG_NOCOOKIE or something.  That's mostly a  cosmetic change, except i think the check for != below would be changed to a check for 0 (no_cookie) and != (reset)?